digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/client/vvh/KgvOperation.scala

```
@@ -0,0 +1,493 @@
+    12312 -> FeatureClass.WinterRoads,
+    12153 -> FeatureClass.SpecialTransportWithoutGate,
+    12154 -> FeatureClass.SpecialTransportWithGate,
+    12131 -> FeatureClass.CarRoad_IIIa,
+    12132 -> FeatureClass.CarRoad_IIIb
+  )
+
+  private val trafficDirectionToTrafficDirection: Map[Int, TrafficDirection] = Map(
+    0 -> TrafficDirection.BothDirections,
+    1 -> TrafficDirection.TowardsDigitizing,
+    2 -> TrafficDirection.AgainstDigitizing)
+
+  private  def extractAdministrativeClass(attributes: Map[String, Any]): AdministrativeClass = {
+    if (attributes("adminclass").asInstanceOf[String] != null)
+      Option(attributes("adminclass").asInstanceOf[String].toInt)
+        .map(AdministrativeClass.apply)
+        .getOrElse(Unknown)
+    else Unknown
+  }
+
+  def extractConstructionType(attributes: Map[String, Any]): ConstructionType = {
+    if (attributes("lifecyclestatus").asInstanceOf[String] != null)
+      Option(attributes("lifecyclestatus").asInstanceOf[String].toInt)
+        .map(ConstructionType.apply)
+        .getOrElse(ConstructionType.InUse)
+    else ConstructionType.InUse
+  }
+  
+  private  def extractTrafficDirection(attributes: Map[String, Any]): TrafficDirection = {
+    if (attributes("directiontype").asInstanceOf[String] != null)
+      Option(attributes("directiontype").asInstanceOf[String].toInt)
+        .map(trafficDirectionToTrafficDirection.getOrElse(_, TrafficDirection.UnknownDirection))
+        .getOrElse(TrafficDirection.UnknownDirection)
+    else TrafficDirection.UnknownDirection
+  }
+  
+  private def extractAttributes(attributesMap: Map[String, Any],validFromDate:BigInt,lastEditedDate:BigInt,starttime:BigInt): Map[String, Any] = {
+    def numberConversio(field:String): BigInt = {
+      try {
+        toBigInt(attributesMap(field).toString.toInt)
+      } catch {
+        case _: Exception =>
+          logger.warn(s"Failed to retrieve value ${field}: ${attributesMap(field)}")
+         0
```

Palautetaanko tässä tarkoituksella nolla virhetilanteissa?

---

Ja funktion nimi numberConversio -> numberConversion

---

hyym kai tässä voisi heittää errorin eikä vaan jatkaa eteenpäin, sellaisella fail fast tyyllisesti

---

nythän me jatketaan eteen päin vaikka jokin conversion epäonnistuisi ajatuskella että silloin kuin kehittäjä testasi homman kaikki toimi ja sen jälkeen tyyppi ei muuttuisi lennosta

---

Ok, se on varmaan parempi lähestymistapa, että linkkien haku ei kaatuisi ns. yhteen konversio virheeseen. Tosin virhetilanteita voi olla vähän hankala huomata suorituksen aikana jos ei tule seurattua lokeja.